### ISSUE TITLE
Comparison with pyimagej

### ISSUE THREAD

**Issue by phisanti** [hanslovsky/imglyb#11]
Hi, 

I write very frequently scripts for Imagej in python using the Jython interpreter but starting to consider using directly python. For that I am considering these options:

- imglyb (your library)
- pyimagej [https://pypi.org/project/pyimagej/]
- https://nbviewer.jupyter.org/github/imagej/tutorials/blob/master/notebooks/ImageJ-Tutorials-and-Demo.ipynb

From the pure scripting approach, is there any significant differences among those?

**Comment by hanslovsky** [comment]
Hi @phisanti, I have left my position in academia and am not working actively on this anymore at this time. 

To answer your question: `imglyb` is a compatibility layer for shared memory between `numpy.ndarray` and ImgLib2 data structures. `pyimagej` uses `imglyb` and also exposes ImageJ + useful helper functions as a CPython library. The difference between those two libraries is that `imglyb` operates at a lower level and `pyimagej` is more comprehensive. I you want to use ImageJ anyways, it is probably best to use `pyimagej`. As for scripting, if you are using ImageJ1 data structures (`ImageProcessor` etc), you will not benefit from the shared memory access, but the CPython environment may provide a better scripting experience than the Jython experience.

The notebooks you linked are executed in the JVM afaik, so they are not CPython compatible but @ctrueden would know for sure.

tldr: If you really want to use CPython, you should probably use `pyimagej`.

cc @ctrueden @imagejan @kephale Please share your scripting experience with `pyimagej` and if you'd recommend it over using Jython (in the context of ImageJ1 or ImageJ2)

**Comment by kephale** [comment]
The short answer is: 
Jython is great if you only need to rely on ImageJ/Fiji/Java projects as dependencies

pyimagej is essentially required if you write your own deep learning tools and need to use ImageJ to use/analyze DL results

**Comment by ctrueden** [comment]
@phisanti In future, please disclose when you cross-post. This was posted also as imagej/pyimagej#76; see my response there.

**Comment by phisanti** [comment]
Thanks for the help and sorry for the not notifying the cross-post! Always learning.

Kind regards
Santi
